### ISSUE TITLE
Weapons Slot

### ISSUE THREAD

**Issue by TopCud** [suriyun-production/mmorpg-kit-docs#2574]
Is there anyway to add more 2 Weapons slot ? for example  1 Right Apparel weapon 1 Left Apparel Weapon

**Comment by insthync** [comment]
Just for character appearance, no action functionality, right?

**Comment by TopCud** [comment]
Yes im gonna use it for weapon skins with the priority system like in armor type

**Comment by insthync** [comment]
For skins?, not something like when equip to right hand then weapon parts are appearing at right hand and right shoulder, right?

**Comment by TopCud** [comment]
For skins yes for example in armor type i can make 2 slots and equip 2 armors one with priority 0 and one with priority 1 then the game will display the higher priority model
Thats what im trying to do with weapons too 
I made a small test with weapon type system i tried to make the left hand slot for skins and right hand slot for actual weapon it worked just perfect and exactly how i want but the problem is this way i will not be able to equip shields or dual weapons thats why im asking if its possible to add more weapons slots

**Comment by Niinra** [comment]
I understand that you want to add more slots for two hand slots and other equipment slots?

**Comment by Niinra** [comment]
Example:
![image](https://github.com/user-attachments/assets/afea70bc-bfaa-48cc-a394-dd679b63ba50)

**Comment by TopCud** [comment]
Exactly slots that will let me equip more weapon type items

**Comment by TopCud** [comment]
> Example:
> ![image](https://private-user-images.githubusercontent.com/115952096/375502488-afea70bc-bfaa-48cc-a394-dd679b63ba50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1ODU2NzQsIm5iZiI6MTcyODU4NTM3NCwicGF0aCI6Ii8xMTU5NTIwOTYvMzc1NTAyNDg4LWFmZWE3MGJjLWJmYWEtNDhjYy1hMzk0LWRkNjc5YjYzYmE1MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxODM2MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMDNlYWI3ZWJhODQzY2NlNGNmMTFlOWE0NTEyZmU5NTdhNTc3MDY2YTIzYWExZDE2ZGI1Njg1YTczMjRiMmM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RpAjax_DoftQZhdxuv12LCgS9saNzi0x38MMjWyqetA)

Not armor type slots but weapon type slots there is difference

**Comment by Niinra** [comment]
Slots that only equip certain types of weapons?

**Comment by TopCud** [comment]
> Slots that only equip certain types of weapons?

Yes

**Comment by Niinra** [comment]
![image](https://github.com/user-attachments/assets/627a51f8-fdba-475a-bf9c-1d9bb4208f4e)
Look for this prefab and review the options that the scripts have available, it is just a matter of reviewing and you will find a way to do what you ask since it is possible

**Comment by TopCud** [comment]
I already checked that sir 
U can add as many type armor slots as u want 
But there is 2 slots only for weapons right hand and left hand 
Adding more require alot of core editing as i seen

**Comment by Niinra** [comment]
![image](https://github.com/user-attachments/assets/bdde8e36-6352-4ea8-a985-2df7677edda1)
You can add more weapon slots as you wish, I have already done so

**Comment by TopCud** [comment]
> ![image](https://private-user-images.githubusercontent.com/115952096/375512896-bdde8e36-6352-4ea8-a985-2df7677edda1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1ODc5MjgsIm5iZiI6MTcyODU4NzYyOCwicGF0aCI6Ii8xMTU5NTIwOTYvMzc1NTEyODk2LWJkZGU4ZTM2LTYzNTItNGVhOC1hOTg1LTJkZjc2NzdlZGRhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxOTEzNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xY2ZhOGVlN2E1YzU2ODg3MDFjMDc2YTViNGYwZGE4NzcxNjQ1YjJlNDRjMDM5ZTM5NjU3ZTM3ODg5ZDU3MjM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qmNg2YKI39OZLykl3uuzWTqbnDKE0L7iN_ufeM3vXz8) You can add more weapon slots as you wish, I have already done so

Sir u are adding more weapon index not weapon slots 
Weapon index add 2 more slots and then u can swap between index 1,2,3 this is not what i asked for :/

**Comment by Niinra** [comment]
Oh, I understood better, you want to add that this slot is only for a specific weapon, you will have to modify the equip items UI code, it is not that difficult to do it and it is not linked to certain specific things in the core, you could override it or create a class complete with what you need

**Comment by TopCud** [comment]
Well if you can do it dm me on discord and we discuss price MiniMan
